library/data/buffer.lean

```diff
@@ -93,7 +93,15 @@ def iterate : Π b : buffer α, β → (fin b.size → α → β → β) → β
 def foreach : Π b : buffer α, (fin b.size → α → α) → buffer α
 | ⟨n, a⟩ f := ⟨n, a.foreach f⟩
 
-def map (f : α → α) : buffer α → buffer α
+/-- Monadically map a function over the buffer. -/
+@[inline]
+def mmap {m} [monad m] (b : buffer α) (f : α → m β) : m (buffer β) := b.2.mmap f
+
+/-- Map a function over the buffer. -/
+@[inline]
+def map (a : buffer n α) (f : α → β) : buffer n β := b.2.map f
+
+def endomap (f : α → α) : buffer α → buffer α
```

Comment:
I'm not sure if this is what Mario had in mind.

Reply:
I'm also not sure... @digama0 what did you have in mind?